### PR TITLE
Revert #2 and #4, update encode/tests

### PR DESCRIPTION
**Pull request by fanatid**

### CODE REVIEW

**Comment by fanatid** [comment]
@dcousens I change conditions in `encode`, need patch (minor?) release.

File: index.js [hunk 1]
```diff
@@ -80,45 +77,29 @@ function decode (buffer) {
...
+  s.copy(signature, 6 + lenR)
 
   return signature
 }
-
-module.exports = {
-  check: check,
-  decode: decode,
-  encode: encode
-}
```
Comment (dcousens) [inline index.js:124]: Personally I prefer the `module.exports` at the end approach... easier to visualize.

**Comment by dcousens** [comment]
Thanks @fanatid

File: index.js (hunk 1 above)
Comment (fanatid) [inline index.js:124]: @dcousens then, let's back to this code. What do you think about `export` right after `require`?

Comment (dcousens) [inline index.js:124]: @fanatid can't stand it.
I follow these three rules which lead to the result I typically have:
- No use before declaration
- No repeated declarations
- Prefer named functions
- Group hackish/lame/ugly code for easy removal one day

Moving the exports to the top breaks my first rule.
The current implementation breaks my second rule,  and removing the function name makes it an anonymous function (ew).
The previous implementation met all these rules.

Ideally,  ES Next's `export`  style syntax would be available... until then,  I use the `module.exports = {}` at EOF.

Thoughts?

Comment (fanatid) [inline index.js:124]: @dcousens thanks for published your rules, it's really helpful.
I'm ok with them, do you need PR for `module.export` or you do this yourself?
Can you publish new version after this?

**Pull request merged**